unity-renderer/Assets/UIComponents/Scripts/Components/BaseComponentView.cs

```diff
@@ -0,0 +1,82 @@
+using System;
+using UnityEngine;
+
+public interface IBaseComponentView
+{
+    /// <summary>
+    /// It will be triggered after the UI component is initialized (it is called on the Start Unity event).
+    /// </summary>
+    event Action OnInitialized;
+
+    /// <summary>
+    /// Returns true if the UI component is already initialized.
+    /// </summary>
+    bool isInitialized { get; }
+
+    /// <summary>
+    /// It is called when the UI component game object awakes.
+    /// </summary>
+    void Initialize();
+
+    /// <summary>
+    /// Shows the UI component.
+    /// </summary>
+    /// <param name="instant">True for not apply progressive animation.</param>
+    void Show(bool instant = false);
+
+    /// <summary>
+    /// Hides the UI component.
+    /// </summary>
+    /// <param name="instant">True for not apply progressive animation.</param>
+    void Hide(bool instant = false);
+
+    /// <summary>
+    /// Updates the UI component with the current model.
+    /// </summary>
+    void RefreshControl();
+
+    /// <summary>
+    /// It is called when the UI component game object is destroyed.
+    /// </summary>
+    void Dispose();
+}
+
+[RequireComponent(typeof(ShowHideAnimator))]
+[RequireComponent(typeof(CanvasGroup))]
+public abstract class BaseComponentView : MonoBehaviour, IBaseComponentView
+{
+    public event Action OnInitialized;
+    public bool isInitialized { get; private set; }
+
+    internal ShowHideAnimator showHideAnimator;
+
+    public virtual void Initialize()
```

Comment:
What I dont like of this pattern is that we enforce inherited classes to call `base.Initialize` or the system will break. 

Maybe this method could be private and have an abstract `Initialize_Impl` or `PreInitialization` `PostInitialization` if we need more granularity in the flow.

Reply:
I like the approach that you suggest, Alex! I have added the `PostInitialization()` method and I have maked the `Initialize()` one as private. In this way, the child classes will not have to worry about calling any base method. Thanks for the feedback! Could you please take a look to the implementation? https://github.com/decentraland/unity-renderer/pull/1175/commits/3eba8846716f0141663a49f20bdf6bcc6b0e0fd0